src/main/java/pageclasses/CartPage.java

```diff
@@ -0,0 +1,112 @@
+package pageclasses;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.TimeoutException;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+import java.util.List;
+
+public class CartPage extends BasePage {
+
+    @FindBy(className = "item-info")
+    private List<WebElement> productList;
+
+    @FindBy(css = ".price .cart-price .price")
+    private List<WebElement> productPrices;
+
+    @FindBy(css = ".input-text.qty")
+    private List<WebElement> productQuantities;
+
+    @FindBy(css = ".subtotal .cart-price .price")
+    private List<WebElement> productSubtotals;
+
+    @FindBy(css = ".sub .price")
+    private WebElement cartSubtotal;
+
+    @FindBy(css = ".grand .price")
+    private WebElement grandTotal;
+
+    private static final By DISCOUNT = By.cssSelector("td[data-th='Discount'] .price");
+    private static final By TAX = By.cssSelector(".totals-tax .price");
```

Comment:
remove By locator. You can use List<WebElement>. Ex:
```
@FindBy(css = "td[data-th='Discount'] .price")
    private List<WebElement> discount;
```
then in the method:
```
if (!discount.isEmpty()) {
            discountValue = Double.parseDouble(discount.get(0).getText()
```
or create method isDisplayed(WebElement element) with `NoSuchElementException` interception
```
catch(NoSuchElementException ex) {
   return false;
```